src/loader.js

```diff
@@ -146,15 +147,186 @@
   };
 
   // 4. Registry Objects
+  // For now, registry objects are a work in progress that don't fully integrate into the rest of the code base
 
   // 4.1.1
+  function createRegistry() {
+    var registry = new RegistryPrototype();
+    registry._registry = {};
+    if (__global.Map)
+      registry._registry.registryMap = new __global.Map();
```

Comment:
Good point. According to https://kangax.github.io/compat-table/es6/, All versions of Safari are okay because they either don't support maps at all (safari 7-) or they support everything we need (safari 8+). However, IE11 is not okay because it supports the Map constructor but not the `entries`, `keys`, or `values` functions which we need. As for Chrome and Firefox, the table doesn't go back far enough to show whether there was ever a period where Map was supported without all of the functions we need on its prototype.

I'll update this so that we feature check for all of the functions that we need on the Map prototype.